src/controllers/vdeployment_controller/trusted/liveness_theorem.rs

```diff
@@ -20,20 +20,36 @@ pub open spec fn vd_eventually_stable_reconciliation_per_cr(vd: VDeploymentView)
 // TODO: add another version which talks about pods and derives from VRS ESR and this ESR
 pub open spec fn current_state_matches(vd: VDeploymentView) -> StatePred<ClusterState> {
     |s: ClusterState| {
-        let objs = s.resources().values().filter(|obj: DynamicObjectView| {
-            &&& obj.kind == VReplicaSetView::kind()
-            &&& obj.metadata.namespace == vd.metadata.namespace
-        }).to_seq();
-        let vrs_list = objects_to_vrs_list(objs);
-        let filtered_vrs_list = filter_vrs_list(vd, vrs_list.unwrap());
-        let (new_vrs, old_vrs_list) = filter_old_and_new_vrs(vd, filtered_vrs_list);
-        &&& vrs_list.is_Some()
+        // simulate handle_list_req
+        let objs = s.resources().values().filter(list_vrs_obj_filter(vd)).to_seq();
+        let (new_vrs, old_vrs_list) = filter_old_and_new_vrs_on_etcd(vd, s.resources());
+        // this step may return None so we need to check here
+        &&& objects_to_vrs_list(objs).is_Some()
         &&& old_vrs_list.len() == 0
+        // TODO: add requirements on owner_ref
         &&& new_vrs.is_Some()
-        &&& new_vrs.unwrap().spec.replicas.unwrap_or(1) == vd.spec.replicas.unwrap_or(1)
         &&& match_template_without_hash(vd, new_vrs.get_Some_0())
+        &&& match_replicas(vd, new_vrs.get_Some_0())
         //&&& current_state_matches(new_vrs_list[0])
     }
 }
 
+pub open spec fn filter_old_and_new_vrs_on_etcd(vd: VDeploymentView, resources: StoredState) -> (Option<VReplicaSetView>, Seq<VReplicaSetView>) {
+    let objs = resources.values().filter(|obj: DynamicObjectView| {
+        &&& obj.kind == VReplicaSetView::kind()
+        &&& obj.metadata.namespace == vd.metadata.namespace
+    }).to_seq();
+    let vrs_list = objects_to_vrs_list(objs);
+    let filtered_vrs_list = filter_vrs_list(vd, vrs_list.unwrap());
+    filter_old_and_new_vrs(vd, filtered_vrs_list)
+}
+
+// why this fn makes proof pass?
```

Comment:
Don't just put a question in the comment. Make it either a TODO for future investigation, or explain why things work.